commands/account/auth.ts

```diff
@@ -0,0 +1,183 @@
+import { setAsDefaultAccountPrompt } from '../../lib/prompts/setAsDefaultAccountPrompt';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { uiFeatureHighlight } from '../../lib/ui';
+import { CommonArgs, ConfigArgs } from '../../types/Yargs';
+
+const i18nKey = 'commands.account.subcommands.auth';
+
+async function createOrUpdateConfig(
+  env: Environment,
+  doesConfigExist: boolean,
+  account?: number
+): Promise<CLIAccount_NEW | null> {
+  try {
+    const { personalAccessKey } = await personalAccessKeyPrompt({
+      env,
+      account,
+    });
+    const token = await getAccessToken(personalAccessKey, env);
+    const defaultName = token.hubName ? toKebabCase(token.hubName) : null;
+
+    const name = doesConfigExist
+      ? undefined
+      : (await cliAccountNamePrompt(defaultName)).name;
+
+    const updatedConfig = await updateConfigWithAccessToken(
+      token,
+      personalAccessKey,
+      env,
+      name,
+      !doesConfigExist
+    );
+
+    if (!updatedConfig) return null;
+
+    if (doesConfigExist && !updatedConfig.name) {
+      updatedConfig.name = (await cliAccountNamePrompt(defaultName)).name;
+    }
+
+    if (doesConfigExist) {
+      updateAccountConfig({
+        ...updatedConfig,
+        environment: updatedConfig.env,
+        tokenInfo: updatedConfig.auth!.tokenInfo,
+      });
+      writeConfig();
+    }
+
+    return updatedConfig as CLIAccount_NEW;
+  } catch (e) {
+    logError(e);
+    return null;
+  }
+}
+
+export const describe = null; // i18n(`${i18nKey}.describe`);
+export const command = 'auth';
+
+type AccountInfoArgs = CommonArgs & ConfigArgs;
+
+export async function handler(
+  args: ArgumentsCamelCase<AccountInfoArgs>
+): Promise<void> {
+  const { providedAccountId, disableTracking } = args;
+
+  if (!disableTracking) {
+    trackCommandUsage('account-auth', {}, providedAccountId);
+  }
+
+  const env = args.qa ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD;
+  const configExists = configFileExists(true);
+
+  if (!configExists) {
+    createEmptyConfigFile({}, true);
+  }
+  loadConfig('');
+
+  handleExit(deleteEmptyConfigFile);
+
+  try {
+    const updatedConfig = await createOrUpdateConfig(
+      env,
+      configExists,
+      providedAccountId
+    );
+
+    if (!updatedConfig) {
+      logger.error(i18n(`${i18nKey}.errors.failedToUpdateConfig`));
+      process.exit(EXIT_CODES.ERROR);
+    }
+
+    const { name, accountId } = updatedConfig;
+
+    // If the config file was just created, we don't need to prompt the user to set as default
+    if (!configExists) {
+      logger.log('');
+      logger.success(
+        i18n(`${i18nKey}.success.configFileCreated`, {
+          configPath: getConfigPath('', true)!,
+        })
+      );
+      logger.success(
+        i18n(`${i18nKey}.success.configFileUpdated`, {
+          account: name || accountId,
+        })
+      );
+    }
+
+    if (configExists) {
```

Comment:
nit, but this could be an `else` block

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1365/commits/2acdfb8eb4d268b18e4e76619c9101f2aa19d8bc